resources/META-INF/plugin.xml

```diff
@@ -48,6 +48,20 @@
                 description="Configure JHelper"
                 icon="/name/admitriev/jhelper/icons/settings.png"
         />
+        <action id="name.admitriev.jhelper.actions.UnarchiveTaskAction"
+                class="name.admitriev.jhelper.actions.UnarchiveTaskAction"
+                text="Restore Tasks"
+                icon="/name/admitriev/jhelper/icons/unarchive.png"
+                description="Restore directory with tasks or a single task back to the former place">
+            <add-to-group group-id="ProjectViewPopupMenu" anchor="first"/>
```

Comment:
Is is possible to add it to menu only for files for which it make sense (e.g cpp&xml & directories)? and ideally only inside project that work with JHelper

Reply:
I thought about it, but it puts a big overhead if we recursively check all the directories and inner files on every menu popup opening. To optimize, we can mark parent directories after the creation of a new JHelper config or a newly archived file, but that looks a bit complicated for now :) What do you think?

As a workaround, we can remove those buttons from the popup menu completely (a user can add them there manually anyway) and have them available via Ctrl+Shift+A 

P.S. Thanks a lot for the very quick reply and for the awesome JHeleper plugin!!

Reply:
I probably would not check recursively, but I'd only add it for cpp&xml files and for directories (if that's possible). Others are guaranteed to not do anything.

Also, I think that adding it to the first position in the menu may be a little to aggressive and we should add it a bit lower.

Workaround you mentioned is always an option, it's what is done with other actions anyway